payments/base/network/src/main/java/com/appcoins/payments/network/RestClientImpl.kt

```diff
@@ -0,0 +1,91 @@
+package com.appcoins.payments.network
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.time.withTimeout
+import kotlinx.coroutines.withContext
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+import java.time.Duration
+
+class RestClientImpl(
+  private val scope: CoroutineScope,
+  private val httpLogger: HttpLogger?,
+  private val baseUrl: String,
+  private val timeout: Duration,
+  private val getUserAgent: GetUserAgent,
+) : RestClient {
+
+  override suspend fun <T : Any> call(
```

Comment:
Do we need to extend "Any" class? Couldn't it be just "T"?

Reply:
Yes, otherwise T will be nullable